File: code/0167.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int start = 0;
        int end = numbers.size()-1;
        while(true){
            if(numbers[start] + numbers[end] == target) break;
            else if(numbers[start] + numbers[end] < target) ++start;
            else --end;
        }
        return vector<int>{start+1,end+1};
    }
};

File: code/0167.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0
        end = len(numbers) - 1
        while start < end:
            if numbers[start] + numbers[end] == target:
                break
            elif numbers[start] + numbers[end] < target:
                start += 1
            else:
                end -= 1
        return [start+1, end+1]